### PR TITLE
[UI] Migrate CustomAvatar and ErrorBoundary components to MUI 5 

### PR DESCRIPTION
**Pull request by Vaibhav701161**
### Note for reviewers:
This PR migrates the `CustomAvatar` and `ErrorBoundary` components to MUI 5, addressing the need to update the UI components in line with the MUI version upgrade.

### Changes Included:
- Migrated the `CustomAvatar` component from MUI 4 to MUI 5:
  - Updated imports from `@material-ui/core` to `@mui/material`.
  - Replaced legacy styling methods with the new `styled` API.
- Migrated the `ErrorBoundary` component from MUI 4 to MUI 5:
  - Updated `Button` imports from `@material-ui/core` to `@mui/material`.
  - Kept the same error handling and fallback structure while ensuring compatibility with MUI 5.

  
Reviewers, please verify that the MUI 5 migration has been implemented correctly, and that the components behave as expected.

- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by SAHU-01** [comment]
Thank you for contributing to this @Vaibhav701161, will review this and notify you!

**Comment by vishalvivekm** [comment]
@Vaibhav701161 
Thank you for your contributions! Let's discuss this during the Meshery Development call today at 9:00 AM CT. Adding it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?pli=1#) || [meeting link](https://meet.layer5.io/meshery).

File: ui/components/CustomAvatar.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+import Badge from '@mui/material/Badge';
+import { styled } from '@mui/material/styles';
 
-const StyledBadge = withStyles((theme) => ({
-  badge: {
-    backgroundColor: ({ color }) => color || '#44b700',
-    color: ({ color }) => color || '#44b700',
+const StyledBadge = styled(Badge)(({ theme, color }) => ({
+  '& .MuiBadge-badge': {
+    backgroundColor: color || '#44b700',
```
Comment (sudhanshutech) [inline ui/components/CustomAvatar.js:7]: use tokens from sistent

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

File: ui/components/CustomAvatar.js [hunk 2]
```diff
@@ -29,23 +29,19 @@ const StyledBadge = withStyles((theme) => ({
...
+    marginRight: theme.spacing(-0.75),
   },
 }));
 
 export default function BadgeAvatars({ children, color }) {
-  const classes = useStyles();
-
   return (
-    <div className={classes.root}>
+    <Root>
```
Comment (sudhanshutech) [inline ui/components/CustomAvatar.js:44]: wrap it in usesSistent wrapper

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

File: ui/components/CustomAvatar.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
 import React from 'react';
-import Badge from '@material-ui/core/Badge';
-import { makeStyles, withStyles } from '@material-ui/core/styles';
+import Badge from '@mui/material/Badge';
```
Comment (sudhanshutech) [inline ui/components/CustomAvatar.js:2]: use imports from sistent because it directly use muiV5.

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

File: ui/components/CustomAvatar.js (hunk 1 above)
Comment (Vaibhav701161) [reply ui/components/CustomAvatar.js]: done

File: ui/components/ErrorBoundary.js [hunk 4]
```diff
@@ -1,41 +1,28 @@
...
+import { Button } from '@mui/material';
 import React from 'react';
+import { ErrorBoundary as SistentErrorBoundary } from '@sistent/react';
 
 /**
  * ErrorBoundary is a React component that catches JavaScript errors in its child components and renders a fallback UI when an error occurs.
  * It should be used as a wrapper around components that might throw errors.
  * @deprecated use error boundary from sistent instead
  */
-class ErrorBoundary extends React.Component {
```
Comment (ianrwhitney) [inline ui/components/ErrorBoundary.js:9]: seems like we can get ride of this comment in the comments now: `@deprecated use error boundary from sistent instead`

Comment (Vaibhav701161) [reply ui/components/ErrorBoundary.js:9]: done

File: ui/components/CustomAvatar.js (hunk 2 above)
Comment (Vaibhav701161) [reply ui/components/CustomAvatar.js:47]: Done

**Comment by SAHU-01** [comment]
Thanks again @Vaibhav701161!
Once you incorporate the sistent components, tokens and styling changes, I believe we should be good.
You can also refer to this [doc](https://docs.meshery.io/project/contributing/contributing-ui-sistent) to understand how to use Sistent

**Pull request closed without merging**